### ISSUE TITLE
GENOME_INIT is redundant?

### ISSUE THREAD

**Issue by jmmcd** [jmmcd/PonyGE2#30]
`GENOME_INIT` seems redundant since we have `INITIALISATION`? `GENOME_INIT` is used only once, in `mapper.py`, when generating an individual from scratch. Maybe `mapper.py` should be calling the initialisation as specified in `params['INITIALISATION']` instead of its own random-genome code.

**Comment by mikefenton** [comment]
I was just talking to Stefan about this. I have updated the readme about initialisation techniques (nearly finished, will commit now in a bit). There are two ways to initialise inds: genome or derivation tree. By default we use smart derivation tree initialisation methods, e.g. RHH. If you specify `--initialisation random_init` then this will by default initialise a population of individuals by generating random derivation trees. However, if you want to initialise using genomes, the only option is to generate random genome strings. Therefore, specifying `--genome_init` will switch from the default derivation tree initialisation techniques to just generating random genome strings. There is no need to specify any `--initialisation` arguments if you are using `--genome_init` as initialisation is set automatically to random (there is no other way to initialise from genomes). 

Does that make sense?

**Comment by jmmcd** [comment]
Hmm... I think it would be more consistent to get rid of `--genome_init` altogether, and use `--initialisation uniform_genome` to get uniformly-distributed genomes.

(I think `random_init` would be better called `uniform_tree` or something like that -- all initialisation methods are random (except for Miguel's new one :) so `random_init` isn't informative!)

**Comment by dvpfagan** [comment]
Again this is the problem that the system is written more like GGGP and not
GE

Random init should make a random genome of size x that gets mapped to a tree

RHH should use a tree to build the desired trees and save a genome.

All inds should have a tree and a genome from initialisation.

Then we could have a --fast_gggp or --fast_genome that strips away the tree
of genome as desired

Anyways just my two cents

**Comment by jmmcd** [comment]
I think the problem is the semantics of the `GENOME_INIT` (and more broadly `GENOME_OPERATIONS`) flags. If I understand it right, the intention is that you can switch these on or off, *independently* of the choice of `INITIALISATION`, `CROSSOVER` (etc) operators. The choice of operators determines the results, but the `GENOME_INIT` and `GENOME_OPERATIONS` can be used to speed things up (if compatible with chosen operators), but they don't affect the results. Right?

But then `GENOME_INIT` is being used wrongly in `mapper.py` line 58, because it's determining the initialisation algorithm.

It would be better to eliminate both `GENOME_INIT` and `GENOME_OPERATIONS` as user parameters. Choose the initialisation algorithm using `INITIALISATION`. Internally, set `GENOME_OPERATIONS` if and only if all of initialisation, mutation and crossover are using genome-oriented operations.

**Comment by mikefenton** [comment]
`GENOME_OPERATIONS` is set automatically, it is not a user-defined option (it's just internal). It just speeds things up dramatically if linear operators are used, it doesn't change the results. 

`GENOME_INIT` _will_ affect the results, as it changes the initialisation method used.

This would be a lot easier to explain in person!

**Comment by jmmcd** [comment]
Let's postpone for now! I'll be there Monday.

**Comment by jmmcd** [comment]
Commit e4eba00 eliminates GENOME_INIT and makes a couple of knock-on changes. Use `--initialisation` instead.
